From 8fed62d57f7f5c99579546459a1c0971a5a3e778 Mon Sep 17 00:00:00 2001
From: Erfan Abdi <erfangplus@gmail.com>
Date: Fri, 21 May 2021 00:59:04 +0430
Subject: [PATCH 03/29] media: Add host media_profiles search paths

---
 media/libmedia/MediaProfiles.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/media/libmedia/MediaProfiles.cpp b/media/libmedia/MediaProfiles.cpp
index eae3c1e..6b8c8fe 100644
--- a/media/libmedia/MediaProfiles.cpp
+++ b/media/libmedia/MediaProfiles.cpp
@@ -48,8 +48,10 @@ std::array<char const*, 5> const& getXmlPaths() {
     static std::array<std::string const, 5> const paths =
         []() -> decltype(paths) {
             // Directories for XML file that will be searched (in this order).
-            constexpr std::array<char const*, 4> searchDirs = {
+            constexpr std::array<char const*, 6> searchDirs = {
                 "product/etc/",
+                "odm_extra/etc/",
+                "vendor_extra/etc/",
                 "odm/etc/",
                 "vendor/etc/",
                 "system/etc/",
-- 
2.55.0

