From b298503693d4cee363f4dd5f702af0a06854d7db Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Sun, 26 Apr 2026 19:45:24 +0800
Subject: [PATCH 20/28] init: Run waydroid-preinit first before doing anything

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 init/first_stage_init.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/init/first_stage_init.cpp b/init/first_stage_init.cpp
index 1836e6c..a92291e 100644
--- a/init/first_stage_init.cpp
+++ b/init/first_stage_init.cpp
@@ -25,6 +25,8 @@
 #include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/utsname.h>
+#include <sys/wait.h>
+#include <spawn.h>
 #include <unistd.h>
 
 #include <chrono>
@@ -338,6 +340,15 @@ static std::unique_ptr<FirstStageMount> CreateFirstStageMount(const std::string&
 }
 
 int FirstStageMain(int argc, char** argv) {
+    // Run waydroid-preinit first before doing anything
+    if (getpid() == 1) {
+        pid_t preinit_pid;
+
+        if (posix_spawn(&preinit_pid, "/system/bin/waydroid-preinit", NULL, NULL, argv, (char *[]){ nullptr }) == 0) {
+            waitpid(preinit_pid, NULL, 0);
+        }
+    }
+
     boot_clock::time_point start_time = boot_clock::now();
 
     std::vector<std::pair<std::string, int>> errors;
-- 
2.55.0

